### DISCUSSION TITLE
винзавод Вино / "Лунный свет"

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#503] [Ideas]
Получается что за винзавод Вино можно получить опыт будучи

- Нейтралом (1 глава)
- Послушником (1 глава)
- Наёмником (1,2 главы)

так?

```
instance DIA_Vino_PERM(C_Info)
{
	npc = BAU_952_Vino;
	nr = 10;
	condition = DIA_Vino_PERM_Condition;
	information = DIA_Vino_PERM_Info;
	permanent = TRUE;
	description = "Есть какие-нибудь интересные новости?";
};


func int DIA_Vino_PERM_Condition()
{
	if(Npc_KnowsInfo(other,DIA_Vino_HALLO) && (Kapitel < 3))
	{
		return TRUE;
	};
};

func void DIA_Vino_PERM_Info()
{
	AI_Output(other,self,"DIA_Vino_PERM_15_00");	//Есть какие-нибудь интересные новости?
	if(Vino_Gossip_Orks == FALSE)
	{
		AI_Output(self,other,"DIA_Vino_PERM_05_01");	//Я могу поклясться, что видел орка вон там, на краю леса, два дня назад.
		AI_Output(self,other,"DIA_Vino_PERM_05_02");	//С тех пор я сплю хотя бы с одним открытым глазом.
		Vino_Gossip_Orks = TRUE;
		Knows_Ork = TRUE;
	}
	else if((Vino_Gossip_Bugs == FALSE) && (MIS_AndreHelpLobart == LOG_Running))
	{
		AI_Output(self,other,"DIA_Vino_PERM_05_03");	//Эти огромные отвратительные насекомые уже всех достали. Они повсюду. Они сожрут все, что мы вырастили, если мы не сделаем что-нибудь.
		AI_Output(self,other,"DIA_Vino_PERM_05_04");	//Несколько дней назад я лежал в траве, размышляя о своем, и собирался вздремнуть, когда одна из этих тварей сожрала мой ботинок!
		AI_Output(self,other,"DIA_Vino_PERM_05_05");	//Ты бы видел, как я улепетывал. С тех пор я не могу спать спокойно.
		Vino_Gossip_Bugs = TRUE;
	}
	else if((FoundVinosKellerei == TRUE) && (Vino_Complain == FALSE) && (hero.guild != GIL_MIL) && (hero.guild != GIL_KDF))
	{
		AI_Output(self,other,"DIA_Vino_PERM45UND6_05_04");	//К сожалению, ополчение нашло мой винокуренный заводик. Надеюсь, что они хотя бы меня не поймают.
		B_GivePlayerXP(150);
		Vino_Complain = TRUE;
	}
	else
	{
		AI_Output(self,other,"DIA_Vino_PERM_05_06");	//Кроме того, что я уже сказал тебе? Нет.
	};
};
```

**Comment by Kor-Angar** [comment]
не так :-)

3 глава Пал и ОНД


```
instance DIA_Vino_KAP3_EXIT(C_Info)
{
	npc = BAU_952_Vino;
	nr = 999;
	condition = DIA_Vino_KAP3_EXIT_Condition;
	information = DIA_Vino_KAP3_EXIT_Info;
	permanent = TRUE;
	description = Dialog_Ende;
};


func int DIA_Vino_KAP3_EXIT_Condition()
{
	if(Kapitel == 3)
	{
		return TRUE;
	};
};

func void DIA_Vino_KAP3_EXIT_Info()
{
	B_NpcClearObsessionByDMT(self);
};


instance DIA_Vino_DMTAMSTART(C_Info)
{
	npc = BAU_952_Vino;
	nr = 31;
	condition = DIA_Vino_DMTAMSTART_Condition;
	information = DIA_Vino_DMTAMSTART_Info;
	permanent = TRUE;
	description = "А как у тебя дела?";
};


func int DIA_Vino_DMTAMSTART_Condition()
{
	if((Kapitel == 3) && (hero.guild != GIL_KDF))
	{
		return TRUE;
	};
};


func void DIA_Vino_DMTAMSTART_Info()
{
	AI_Output(other,self,"DIA_Vino_DMTAMSTART_15_00");	//А как у тебя дела?
	if((FoundVinosKellerei == TRUE) && (Vino_Complain == FALSE) && (hero.guild != GIL_MIL))
	{
		AI_Output(self,other,"DIA_Vino_DMTAMSTART_05_01");	//Дерьмово. Ополчение нашло мой тайный винокуренный заводик.
		AI_Output(self,other,"DIA_Vino_DMTAMSTART_05_02");	//Остается надеяться, что они никогда не узнают, что он принадлежал мне.
		B_GivePlayerXP(150);
		Vino_Complain = TRUE;
	}
	else
	{
		AI_Output(self,other,"DIA_Vino_DMTAMSTART_05_03");	//Эти ублюдки в черных рясах повсюду. Кто-нибудь должен засунуть их капюшоны им в глотку.
	};
};

```

**Comment by Kor-Angar** [comment]
4,5 главы

ОНД и Пал 

```
instance DIA_Vino_PERM45UND6(C_Info)
{
	npc = BAU_952_Vino;
	nr = 42;
	condition = DIA_Vino_PERM45UND6_Condition;
	information = DIA_Vino_PERM45UND6_Info;
	permanent = TRUE;
	description = "Есть новости?";
};


func int DIA_Vino_PERM45UND6_Condition()
{
	if((Kapitel >= 4) && (hero.guild != GIL_KDF) && (NpcObsessedByDMT_Vino == FALSE))
	{
		return TRUE;
	};
};

func void DIA_Vino_PERM45UND6_Info()
{
	AI_Output(other,self,"DIA_Vino_PERM45UND6_15_00");	//Есть новости?
	if((FoundVinosKellerei == TRUE) && (Vino_Complain == FALSE) && (hero.guild != GIL_MIL))
	{
		AI_Output(self,other,"DIA_Vino_PERM45UND6_05_04");	//К сожалению, ополчение нашло мой винокуренный заводик. Надеюсь, что они хотя бы меня не поймают.
		B_GivePlayerXP(150);
		Vino_Complain = TRUE;
	}
	else if(hero.guild == GIL_PAL)
	{
		AI_Output(self,other,"DIA_Vino_PERM45UND6_05_01");	//Здесь все просто кишит орками.
		AI_Output(self,other,"DIA_Vino_PERM45UND6_05_02");	//Вы, паладины, ведь прикончите их, да?
	}
	else
	{
		AI_Output(self,other,"DIA_Vino_PERM45UND6_05_03");	//Боюсь, что скоро нам придется покинуть нашу ферму и присоединиться к наемникам на ферме Онара.
	};
};
```

**Comment by Kor-Angar** [comment]
итоговый такой список?

Нейтралом (1-5 глава)
Послушником (1-5 глава)
Наёмником (1-5 главы)
ОНД (3-5 главы)
Пал (3-5 главы)

**Comment by Kor-Angar** [comment]
наверное Паладину так же не следует рассказывать про винзавод.

**Comment by dosinabox** [comment]
Установка переменной Vino_Complain в значение TRUE обеспечивает получение 150 единиц опыта только один раз.

> наверное Паладину так же не следует рассказывать про винзавод.

Думаю, паладинам пофиг на городские дела. Как и магам.
Получается, что опыт могут получить все, кроме ополченцев (Вино не скажет об этом) и магов (для них новости заменяются на одержимость и последующее лечение).

**Comment by Kor-Angar** [comment]
>  магов (для них новости заменяются на одержимость и последующее лечение).

во второй главе нет.

с паладином получается так: Вино не говорит герою метну, а потом когда герой становится Паладином неожиданно рассказывает.

Маги, Менты и Паладины это же представители власти. а рассказ Вино о противозаконной истории. т.е. получается что Вино признаётся Паладину о преступление.

**Comment by Kor-Angar** [comment]
так :-)
в Готике 2 оригинал и в Готике  2 НВ есть квест про винзавод "Лунный свет", он доступен с 3 главы. Магу огня он недоступен

```
instance DIA_Ehnim_MoleRatFett(C_Info)
{
	npc = BAU_944_Ehnim;
	condition = DIA_Ehnim_MoleRatFett_Condition;
	information = DIA_Ehnim_MoleRatFett_Info;
	important = TRUE;
};


func int DIA_Ehnim_MoleRatFett_Condition()
{
	if((DIA_Ehnim_STREIT5_noPerm == TRUE) && (Kapitel >= 3) && (hero.guild != GIL_KDF))
	{
		return TRUE;
	};
};
```
в Ночь Ворона "жир кротокрыса" добавили в продажу Мартину с 1 главы. 

[в Готика 2 классик сдать квест Вино можно только с 3 главы.
](https://github.com/dosinabox/g2_classic/blob/master/PrjGothic/Story/Dialoge/DIA_bau_952_vino.d)в Готике 2 Ночь Ворона сдать квест можно с первой главы.

в связи с этим предлагается ещё раз посмотреть оригинальный код Вино из Готике 2 НВ, что бы убедится в том что эти изменения авторские.

если изменения авторские тогда можно снять ограничения на квест "Лунный свет" у Ehnim на 3 главу, предположив что авторы поправили Вино, но забыли поправить у Ehnim.

так же можно добавить возможность красть у Ehnim "жир кротокрыса", вместо денег.

**Comment by dosinabox** [comment]
> в связи с этим предлагается ещё раз посмотреть оригинальный код Вино из Готике 2 НВ, что бы убедится в том что эти изменения авторские.

Посмотреть оригинальный код Г2НВ можно по этой ссылке:
https://github.com/dosinabox/g2nr_unofficial_update/tree/98e37682c9eaece0a90dfd2aec76977c1a7656a9

**Comment by Kor-Angar** [comment]
> в связи с этим предлагается ещё раз посмотреть оригинальный код Вино из Готике 2 НВ, что бы убедится в том что эти изменения авторские.

> Посмотреть оригинальный код Г2НВ можно по этой ссылке

в оригинальных скриптах  Г2НВ квест Вино нельзя сдать в 1 и 2 главах
![image](https://user-images.githubusercontent.com/49514764/72815891-69faaf00-3c78-11ea-93a5-6ec99ecbb645.png)

**Comment by Kor-Angar** [comment]
в связи с этим есть несколько вариантов:

- [ ] довести неавторскую доделку до логического конца, сняв ограничение на 3 главу в разговоре с Ehnim, таким образом квест можно будет не только "сдавать" Вино, но и "брать" у Ehnim. Сейчас квест сдаётся  с 1 главы,  а берётся с 3 главы.
- [ ] откатится на оригинальный скрипт, дав возможность проходить этот квест только с 3 главы
- [ ] ничего не делать

кстати [из самой книги Вино](https://gothic.ag.ru/downloads/images/g2/vino/Sc_2.jpg) вовсе не следует что хозяин винзавода Вино. Об этом герой узнаёт от квестодателя Ehnim

> _//Ты ВСЕ ЕЩЕ здесь?
> //Ну да. Все еще ссоритесь?
> //Не твое дело. Забудь об этом. Скажи, ты не был недавно на ферме Лобарта?
> //Может быть. А что?
> //Ох, ничего такого. Я просто хотел поговорить с Вино о его винокурне.
> //Винокурня? Какая винокурня?
> //Ох. Мне, наверное, не стоило говорить об этом. Вино всегда так трепетно относился к этому своему маленькому секрету.
> //Ну ладно, все равно уже я проболтался. В лесу, неподалеку от фермы, Вино устроил секретную винокурню.
> //Несколько дней назад он попросил меня дать ему что-нибудь, чтобы смазать механизм решетки, установленной там.
> //Последнее время шли дожди, и она заржавела. Лебедку заклинило, и он не может поднять решетку._

поэтому было бы логично в сдаче квеста Вино добавить проверку на то взят ли квест(т.е. узнал ли ГГ от Ehnim, что винзавод принадлежит Вино).

*****

Тут ещё важный момент, возможность попасть в винзавод в 1 главе была даже в Готике 2 Классик, т.к. жир был в трюме корабля. а вот получить квест и сдать его можно было только с 3 главы. в НВ добавили ещё один жир в продажу Мартину. т.е. по большому счёту в НВ ничего не поменялось :-)

**Comment by Kor-Angar** [comment]
нужно делать хоть какие то записи в дневник (TOPIC_FoundVinosKellerei) :

- Открытие пещеры
- Прочтение книги

**Comment by Kor-Angar** [comment]
информацию от Ehnim лучше разбить на две записи
```D
func void DIA_Ehnim_MoleRatFett_was()
{
	AI_Output(other,self,"DIA_Ehnim_MoleRatFett_was_15_00");	//Винокурня? Какая винокурня?
	AI_Output(self,other,"DIA_Ehnim_MoleRatFett_was_12_01");	//Ох. Мне, наверное, не стоило говорить об этом. Вино всегда так трепетно относился к этому своему маленькому секрету.
	AI_Output(self,other,"DIA_Ehnim_MoleRatFett_was_12_02");	//Ну ладно, все равно уже я проболтался. В лесу, неподалеку от фермы, Вино устроил секретную винокурню.
	AI_Output(self,other,"DIA_Ehnim_MoleRatFett_was_12_03");	//Несколько дней назад он попросил меня дать ему что-нибудь, чтобы смазать механизм решетки, установленной там.
	AI_Output(self,other,"DIA_Ehnim_MoleRatFett_was_12_04");	//Последнее время шли дожди, и она заржавела. Лебедку заклинило, и он не может поднять решетку.
	Log_CreateTopic(TOPIC_FoundVinosKellerei,LOG_MISSION);
	Log_SetTopicStatus(TOPIC_FoundVinosKellerei,LOG_Running);
	B_LogEntry(TOPIC_FoundVinosKellerei,"Эним рассказал мне, что Вино содержит секретный винокуренный завод в лесу около фермы Акила.");
	if(FoundVinosKellerei == FALSE)
	{
		B_LogEntry(TOPIC_FoundVinosKellerei," Но механизм, открывающий решетку, заклинило, и починить его можно только смазав шестеренки жиром крысокрота.");
		Info_AddChoice(DIA_Ehnim_MoleRatFett,"И? Ты достал смазку?",DIA_Ehnim_MoleRatFett_was_Fett);
	}
	else
	{
		DIA_Common_Yeah();	//Ну да.
	};
};
```

**Comment by Kor-Angar** [comment]
для квеста надо какую то переменную ввести, что бы адекватно делать проверки. например `mis_foundVinosKellerei`

**Reply by Kor-Angar** [reply]
- `mis_foundVinosKellerei = LOG_SUCCESS;` у Вино вместо `Vino_Complain = TRUE;`
- `mis_foundVinosKellerei = LOG_Running;` в функции `DIA_Ehnim_MoleRatFett_was`
- `mis_foundVinosKellerei = LOG_Running;` в функции `evt_vinoskellerei_func_s1`
- `mis_foundVinosKellerei = LOG_FAILED;` в функции `B_CheckDeadMissionNPCs`
